pkg/controller/pxc/upgrade.go

```diff
@@ -26,9 +26,9 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 	currentSet.Spec.Replicas = &podSpec.Size
 
 	switch cr.Spec.UpdateStrategy {
-	case "OnDelete":
+	case "OnDelete", "SmartUpdate":
 		currentSet.Spec.UpdateStrategy = appsv1.StatefulSetUpdateStrategy{}
-		currentSet.Spec.UpdateStrategy.Type = cr.Spec.UpdateStrategy
+		currentSet.Spec.UpdateStrategy.Type = appsv1.OnDeleteStatefulSetStrategyType
 	default:
 		currentSet.Spec.UpdateStrategy.Type = cr.Spec.UpdateStrategy
```

Comment:
I think it will be a merge conflict here

Reply:
Will be fixed when K8SPXC-257 approved